components/sup/src/event/nats_message_stream.rs

```diff
@@ -1,30 +1,116 @@
-use crate::event::{Error,
-                   EventStreamConfig,
+use crate::event::{EventStreamConfig,
                    Result};
 use futures::{channel::{mpsc as futures_mpsc,
                         mpsc::UnboundedSender},
               stream::StreamExt};
-use rants::{error::Error as RantsError,
-            native_tls::TlsConnector,
-            Client,
-            Subject};
-use tokio::time;
+use nats::{self,
+           asynk::{Options, Connection}};
+use tokio::{time, sync::Mutex};
+use std::{
+    io,
+    path::PathBuf,
+    sync::Arc,
+    time::Duration};
 
 /// The subject and payload of a NATS message.
 #[derive(Debug)]
 pub struct NatsMessage {
-    subject: &'static Subject,
+    subject: &'static str,
     payload: Vec<u8>,
 }
 
 impl NatsMessage {
-    pub fn new(subject: &'static Subject, payload: Vec<u8>) -> Self {
+    pub fn new(subject: &'static str, payload: Vec<u8>) -> Self {
```

Comment:
`&'static str` to `&str`

Reply:
Pending as related to comment:
https://github.com/habitat-sh/habitat/pull/8293#discussion_r712062829